code/search/README.md

```diff
@@ -1,2 +1,14 @@
-# cosmos
-Your personal library of every algorithm and data structure code that you will ever encounter
+/* Part of Cosmos by OpenGenus Foundation */
+
+## Search algorithms
+
+In computer science, a search algorithm is any algorithm which solves the Search problem, namely, to retrieve information stored within some data structure, or calculated in the search space of a problem domain.
+The appropriate search algorithm often depends on the data structure being searched, but also on any a prior knowledge about the data.
+
+Search algorithms can be classified based on their mechanism of searching. Linear search algorithms check every record for the one associated with a target key in a linear fashion. Binary, or half interval searches, repeatedly target the center of the search structure and divide the search space in half. Comparison search algorithms improve on linear searching by successively eliminating records based on comparisons of the keys until the target record is found, and can be applied on data structures with a defined order. Digital search algorithms work based on the properties of digits in data structures that use numerical keys. Finally, hashing directly maps keys to records based on a hash function. Searches outside of a linear search require that the data be sorted in some way.
+
+Search functions are also evaluated on the basis of their complexity, or maximum theoretical run time.
+Binary search functions, for example, have a maximum complexity of O(log n), or logarithmic time. 
```

Comment:
The term _maximum complexity_ is wrong and is never used. You might want to use _average case complexity_ 👍